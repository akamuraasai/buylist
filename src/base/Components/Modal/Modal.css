.Modal {
    width: 100%;
    height: 100%;
    position: fixed;
    top: 0;
    background-color: rgba(41, 41, 41, 0.38);
    z-index: 1;
    opacity: 0;
    pointer-events: none;
    transition: all 0.3s;
}

.Modal__content {
    width: 90%;
    height: 390px;
    top: 20%;
    position: relative;
    margin: 5%;
    padding: 2rem;
    background: #fff;
    color: #444;
}

.Modal__content > * > i {
    color: #4183c4;
}

.Modal__content__name {
    width: 230px !important;
}

.Modal__content__completed {
    display: none;
}

input[type="checkbox"] + label span {
    display:inline-block;
    width:19px;
    height:19px;
    margin:-2px 10px 0 0;
    vertical-align:middle;
    background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/210284/check_radio_sheet.png) left top no-repeat;
    cursor:pointer;
}

input[type="checkbox"]:checked + label span {
    background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/210284/check_radio_sheet.png) -19px top no-repeat;
}

.Modal__content > * > input[type=text] {
    -webkit-appearance: textfield;
    -moz-appearance: textfield;
    width: 70px;
}

.Modal__close {
    float: right;
}

.Modal__button {
    width: 100%;
    padding: 12px;
    font-size: 16px;
    color: #131313;
    background-color: #b2c5ff;
    border-radius: 5px;
    box-shadow: 1px 1px 1px 0 #c1c1c1;
}

.Modal:target {
    opacity: 1;
    pointer-events: auto;
}