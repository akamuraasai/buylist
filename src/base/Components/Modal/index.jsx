import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { changeField, apply, remove } from '../../Collection/ProductList/ProductList.actions';

import './Modal.css';

const applyChanges = (item, fn) => {
  fn(item);
  window.location = '#';
};

const Modal = ({ item, name, price, units, unitType, completed, changeField, apply, remove }) => (
  <div id="modal" className="Modal">
    <div className="Modal__content">
      <a href="#" title="Close" className="Modal__close">
        <i className="fa fa-close"/>
      </a>
      <h1>
        <input
          type="checkbox"
          className="Modal__content__completed" id="completed"
          checked={completed}
          onClick={() => changeField('completed', !completed)}
        />
        <label htmlFor="completed"><span></span></label>
        <input
          type="text"
          value={name}
          className="Modal__content__name"
          onChange={e => changeField('name', e.target.value)}
        />
        {item.id !== undefined && <i className="fa fa-trash" onClick={() => applyChanges(item.id, remove)} />}
      </h1>

      <b>Preço Unitário</b>
      <p>R$ <input type="text" value={price} onChange={e => changeField('price', e.target.value)} /></p>

      <b>Unidades</b>
      <p><input type="text" value={units} onChange={e => changeField('units', e.target.value)} /></p>

      <b>Tipo</b>
      <p><input type="text" value={unitType} onChange={e => changeField('unitType', e.target.value)} /></p>

      <b>Total</b>
      <p>R$ {(price * units)}</p>

      <button className="Modal__button" onClick={() => applyChanges(item, apply)}>Aplicar Alterações</button>
    </div>
  </div>
);

const mapState2Props = state => {
  const { item } = state.products;
  return {
    item: item,
    name: item.name,
    price: item.price,
    units: item.units,
    unitType: item.unitType,
    completed: item.completed,
  };
};

const mapDispatch2Props = dispatch => bindActionCreators({ changeField, apply, remove }, dispatch);

export default connect(mapState2Props, mapDispatch2Props)(Modal);
