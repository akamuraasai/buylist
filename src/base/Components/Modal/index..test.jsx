import React from 'react';
import { shallow } from 'enzyme';
import Modal from './index';

describe('Modal', () => {
  const component = shallow(<Modal />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });
});
