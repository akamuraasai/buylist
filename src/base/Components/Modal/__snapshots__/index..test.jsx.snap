// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="Modal__content"
>
        <a
                className="Modal__close"
                href="#modal-close"
                title="Close"
        >
                <i
                        className="fa fa-close"
                />
        </a>
        <h1>
                <input
                        className="Modal__content__completed"
                        defaultChecked={undefined}
                        id="completed"
                        onClick={[Function]}
                        type="checkbox"
                />
                <label
                        htmlFor="completed"
                >
                        <span />
                </label>
                <input
                        className="Modal__content__name"
                        type="text"
                        value={undefined}
                />
        </h1>
        <b>
                Preço Unitário
        </b>
        <p>
                R$ 
                <input
                        type="text"
                        value={undefined}
                />
        </p>
        <b>
                Unidades
        </b>
        <p>
                <input
                        type="text"
                        value={undefined}
                />
        </p>
        <b>
                Tipo
        </b>
        <p>
                <input
                        type="text"
                        value={undefined}
                />
        </p>
        <b>
                Total
        </b>
        <p>
                R$ 
                NaN
        </p>
        <button
                className="Modal__button"
        >
                Aplicar Alterações
        </button>
</div>,
      "className": "Modal",
      "id": "modal",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="Modal__close"
            href="#modal-close"
            title="Close"
>
            <i
                        className="fa fa-close"
            />
</a>,
          <h1>
            <input
                        className="Modal__content__completed"
                        defaultChecked={undefined}
                        id="completed"
                        onClick={[Function]}
                        type="checkbox"
            />
            <label
                        htmlFor="completed"
            >
                        <span />
            </label>
            <input
                        className="Modal__content__name"
                        type="text"
                        value={undefined}
            />
</h1>,
          <b>
            Preço Unitário
</b>,
          <p>
            R$ 
            <input
                        type="text"
                        value={undefined}
            />
</p>,
          <b>
            Unidades
</b>,
          <p>
            <input
                        type="text"
                        value={undefined}
            />
</p>,
          <b>
            Tipo
</b>,
          <p>
            <input
                        type="text"
                        value={undefined}
            />
</p>,
          <b>
            Total
</b>,
          <p>
            R$ 
            NaN
</p>,
          <button
            className="Modal__button"
>
            Aplicar Alterações
</button>,
        ],
        "className": "Modal__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa fa-close"
/>,
            "className": "Modal__close",
            "href": "#modal-close",
            "title": "Close",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa fa-close",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="Modal__content__completed"
                defaultChecked={undefined}
                id="completed"
                onClick={[Function]}
                type="checkbox"
/>,
              <label
                htmlFor="completed"
>
                <span />
</label>,
              <input
                className="Modal__content__name"
                type="text"
                value={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Modal__content__completed",
                "defaultChecked": undefined,
                "id": "completed",
                "onClick": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span />,
                "htmlFor": "completed",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "span",
              },
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "Modal__content__name",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Preço Unitário",
          },
          "ref": null,
          "rendered": "Preço Unitário",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R$ ",
              <input
                type="text"
                value={undefined}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "R$ ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Unidades",
          },
          "ref": null,
          "rendered": "Unidades",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              type="text"
              value={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Tipo",
          },
          "ref": null,
          "rendered": "Tipo",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              type="text"
              value={undefined}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Total",
          },
          "ref": null,
          "rendered": "Total",
          "type": "b",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "R$ ",
              NaN,
            ],
          },
          "ref": null,
          "rendered": Array [
            "R$ ",
            NaN,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Aplicar Alterações",
            "className": "Modal__button",
          },
          "ref": null,
          "rendered": "Aplicar Alterações",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="Modal__content"
>
          <a
                    className="Modal__close"
                    href="#modal-close"
                    title="Close"
          >
                    <i
                              className="fa fa-close"
                    />
          </a>
          <h1>
                    <input
                              className="Modal__content__completed"
                              defaultChecked={undefined}
                              id="completed"
                              onClick={[Function]}
                              type="checkbox"
                    />
                    <label
                              htmlFor="completed"
                    >
                              <span />
                    </label>
                    <input
                              className="Modal__content__name"
                              type="text"
                              value={undefined}
                    />
          </h1>
          <b>
                    Preço Unitário
          </b>
          <p>
                    R$ 
                    <input
                              type="text"
                              value={undefined}
                    />
          </p>
          <b>
                    Unidades
          </b>
          <p>
                    <input
                              type="text"
                              value={undefined}
                    />
          </p>
          <b>
                    Tipo
          </b>
          <p>
                    <input
                              type="text"
                              value={undefined}
                    />
          </p>
          <b>
                    Total
          </b>
          <p>
                    R$ 
                    NaN
          </p>
          <button
                    className="Modal__button"
          >
                    Aplicar Alterações
          </button>
</div>,
        "className": "Modal",
        "id": "modal",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="Modal__close"
              href="#modal-close"
              title="Close"
>
              <i
                            className="fa fa-close"
              />
</a>,
            <h1>
              <input
                            className="Modal__content__completed"
                            defaultChecked={undefined}
                            id="completed"
                            onClick={[Function]}
                            type="checkbox"
              />
              <label
                            htmlFor="completed"
              >
                            <span />
              </label>
              <input
                            className="Modal__content__name"
                            type="text"
                            value={undefined}
              />
</h1>,
            <b>
              Preço Unitário
</b>,
            <p>
              R$ 
              <input
                            type="text"
                            value={undefined}
              />
</p>,
            <b>
              Unidades
</b>,
            <p>
              <input
                            type="text"
                            value={undefined}
              />
</p>,
            <b>
              Tipo
</b>,
            <p>
              <input
                            type="text"
                            value={undefined}
              />
</p>,
            <b>
              Total
</b>,
            <p>
              R$ 
              NaN
</p>,
            <button
              className="Modal__button"
>
              Aplicar Alterações
</button>,
          ],
          "className": "Modal__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fa fa-close"
/>,
              "className": "Modal__close",
              "href": "#modal-close",
              "title": "Close",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fa fa-close",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="Modal__content__completed"
                  defaultChecked={undefined}
                  id="completed"
                  onClick={[Function]}
                  type="checkbox"
/>,
                <label
                  htmlFor="completed"
>
                  <span />
</label>,
                <input
                  className="Modal__content__name"
                  type="text"
                  value={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "Modal__content__completed",
                  "defaultChecked": undefined,
                  "id": "completed",
                  "onClick": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span />,
                  "htmlFor": "completed",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "span",
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "Modal__content__name",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Preço Unitário",
            },
            "ref": null,
            "rendered": "Preço Unitário",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "R$ ",
                <input
                  type="text"
                  value={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "R$ ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Unidades",
            },
            "ref": null,
            "rendered": "Unidades",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                type="text"
                value={undefined}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Tipo",
            },
            "ref": null,
            "rendered": "Tipo",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                type="text"
                value={undefined}
/>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Total",
            },
            "ref": null,
            "rendered": "Total",
            "type": "b",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "R$ ",
                NaN,
              ],
            },
            "ref": null,
            "rendered": Array [
              "R$ ",
              NaN,
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Aplicar Alterações",
              "className": "Modal__button",
            },
            "ref": null,
            "rendered": "Aplicar Alterações",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
